File: py/cli/src/tuish_cli/commands/keys.py
from __future__ import annotations

import typer
from rich.console import Console

from tuish_cli.config import load_config, require_api_key
from tuish_cli.output import emit_json

console = Console()


def keys(ctx: typer.Context) -> None:
    config = load_config(ctx.obj.config_path)
    api_key = require_api_key(ctx.obj, config)

    if ctx.obj.json:
        payload = {"apiKey": api_key}
        if config.api_base_url:
            payload["apiBaseUrl"] = config.api_base_url
        emit_json(payload)
        return

    console.print("[bold]API Key[/bold]")
    console.print(api_key)

    if config.api_base_url:
        console.print()
        console.print("[bold]API Base URL[/bold]")
        console.print(config.api_base_url)

File: rs/cli/src/commands/keys.rs
use super::{load_config, output_json, require_api_key, CliResult, Context};
use colored::Colorize;
use serde_json;

pub fn run(ctx: &Context) -> CliResult {
	let config = load_config(&ctx.config_path)?;
	let api_key = require_api_key(&config)?;

	if ctx.json {
		let payload = serde_json::json!({
			"apiKey": api_key,
			"apiBaseUrl": config.api_base_url,
		});
		return output_json(&payload);
	}

	println!("{}", "API Key".bold());
	println!("{api_key}");

	if let Some(url) = config.api_base_url {
		println!();
		println!("{}", "API Base URL".bold());
		println!("{url}");
	}
	Ok(())
}
